web_sdk/web_test_utils/lib/skia_client.dart

```diff
@@ -0,0 +1,399 @@
+      _goldctl,
+      'imgtest', 'init',
+      '--instance', _instance,
+      '--work-dir', _tempPath,
+      '--commit', commitHash,
+      '--keys-file', keys.path,
+      '--failure-file', failures.path,
+      '--passfail',
+      '--crs', 'github',
+      '--patchset_id', commitHash,
+      ...getCIArguments(),
+    ];
+
+    if (imgtestInitCommand.contains(null)) {
+      final StringBuffer buf = StringBuffer()
+        ..writeln('A null argument was provided for Skia Gold tryjob init.')
+        ..writeln('Please confirm the settings of your golden file test.')
+        ..writeln('Arguments provided:');
+      imgtestInitCommand.forEach(buf.writeln);
+      throw Exception(buf.toString());
+    }
+
+    final ProcessResult result = await process.run(imgtestInitCommand);
+
+    if (result.exitCode != 0) {
+      final StringBuffer buf = StringBuffer()
+        ..writeln('Skia Gold tryjobInit failure.')
+        ..writeln('An error occurred when initializing golden file tryjob with ')
+        ..writeln('goldctl.')
+        ..writeln()
+        ..writeln('Debug information for Gold:')
+        ..writeln('stdout: ${result.stdout}')
+        ..writeln('stderr: ${result.stderr}');
+      throw Exception(buf.toString());
+    }
+    _isInitialized = true;
+  }
+
+  /// Executes the `imgtest add` command in the `goldctl` tool for tryjobs.
+  ///
+  /// The `imgtest` command collects and uploads test results to the Skia Gold
+  /// backend, the `add` argument uploads the current image test. A response is
+  /// returned from the invocation of this command that indicates a pass or fail
+  /// result for the tryjob.
+  ///
+  /// The [testName] and [goldenFile] parameters reference the current
+  /// comparison being evaluated.
+  Future<void> tryjobAdd(String testName, File goldenFile) async {
+    final List<String> imgtestCommand = <String>[
+      _goldctl,
+      'imgtest', 'add',
+      '--work-dir', _tempPath,
+      '--test-name', cleanTestName(testName),
+      '--png-file', goldenFile.path,
+    ];
+
+    final ProcessResult result = await process.run(imgtestCommand);
+
+    final String resultStdout = result.stdout.toString();
+    if (result.exitCode != 0 &&
+      !(resultStdout.contains('Untriaged') || resultStdout.contains('negative image'))) {
+      final StringBuffer buf = StringBuffer()
+        ..writeln('Unexpected Gold tryjobAdd failure.')
+        ..writeln('Tryjob execution for golden file test $testName failed for')
+        ..writeln('a reason unrelated to pixel comparison.')
+        ..writeln()
+        ..writeln('Debug information for Gold:')
+        ..writeln('stdout: ${result.stdout}')
+        ..writeln('stderr: ${result.stderr}')
+        ..writeln();
+      throw Exception(buf.toString());
+    }
+  }
+
+  /// Returns the latest positive digest for the given test known to Skia Gold
+  /// at head.
+  Future<String?> getExpectationForTest(String testName) async {
```

Comment:
This is only being used in the 'are we online?' check above. The framework uses this method to do that, but it mainly uses this to get image info for local testing. Is all of this necessary if it's only being used to check online status?

Reply:
I copied this entire file from the framework repo (with some modifications). I didn't want to pick and choose what to copy, especially that we are going to need this method and `getImageBytes` once we start supporting local testing using Skia Gold.